packages/server/modules/auth/services/mailchimp.ts

```diff
@@ -72,21 +72,22 @@ async function updateMailchimpMemberTags(
   initializeMailchimp()
   const subscriberHash = md5(user.email.toLowerCase())
 
-  // First ensure user is in the audience
-  await addToMailchimpAudience(user, listId)
+  // Check if user is already in audience (meaning they consented to marketing emails)
+  try {
+    await mailchimp.lists.getListMember(listId, subscriberHash)
+  } catch (e) {
+    return
```

Comment:
this should be an actual error no?
we do expect the user to be added to the onboarding list